common/persistence/elasticsearch/esVisibilityMetricClients.go

```diff
@@ -77,6 +77,20 @@ func (p *visibilityMetricsClient) RecordWorkflowExecutionClosed(request *p.Recor
 	return err
 }
 
+func (p *visibilityMetricsClient) UpsertWorkflowExecution(request *p.UpsertWorkflowExecutionRequest) error {
```

Comment:
UpsertSearchAttributes?

Reply:
Reason using UpsertWorkflowExecution as interface name instead of UpsertSearchAttributes is: on persistence layer, it actually put the whole execution and doesn't need to be constraint to search attributes,  I see opportunities to reuse this and refactor some existing code